admin-frontend/app_singleapp/lib/widgets/setup/page1_widget.dart

```diff
@@ -238,31 +248,38 @@ class _SetupPage1ThirdPartyProviders extends StatelessWidget {
       required this.bloc,
       required this.selectedExternalProviderFunc})
       : super(key: key);
-
   @override
   Widget build(BuildContext context) {
     final external = bloc.has3rdParty;
-
-    final children = <Widget>[];
-
     if (external) {
-      bloc.externalProviders.forEach((provider) {
-        children.add(InkWell(
-          mouseCursor: SystemMouseCursors.click,
-          onTap: () {
-            selectedExternalProviderFunc(provider);
-          },
-          child: Image.asset(bloc.externalProviderAssets[provider]!),
-        ));
-        children.add(Padding(
-          padding: const EdgeInsets.fromLTRB(0, 16, 0, 10),
-          child: Text('or enter your details to register',
-              style: Theme.of(context).textTheme.caption),
-        ));
-      });
+      return Column(
+        children: [
+          for (dynamic provider in bloc.externalProviders)
+            Container(
+              height: 48,
+              child: Padding(
+                padding: const EdgeInsets.only(top: 12.0),
+                child: SignInButton(
+                    provider == 'oauth2-google'
```

Comment:
can you turn this into a method? We will have a few more coming and this isn't a very scalable way of doing it. Maybe a class that defines the button and the message for register vs login?

Reply:
I thought about it, but this is the simplest way, there is no point to overcomplicate I think with additional abstractions at this point.  "Sign in" message is a default one, so I don't have to provide it.